src/interface/state_descriptor.cpp

```diff
@@ -131,54 +123,54 @@ class DependencyTracker {
   std::unordered_set<std::string> provided_vars;
   std::unordered_set<std::string> depends_vars;
 
-  Dictionary<int> overridable_vars;
-  Dictionary<std::unordered_map<int, bool>> sparse_added_provides;
-  Dictionary<std::unordered_map<int, bool>> sparse_added_depends;
-  Dictionary<std::vector<Metadata>> overridable_meta;
+  std::unordered_map<std::string, int> overridable_vars;
+  std::unordered_map<std::string, std::vector<Metadata>> overridable_meta;
 };
 
 // Helper functions for adding vars
 // closures by reference
-class CellVarProvider : public VariableProvider {
+class DenseFieldProvider : public VariableProvider {
  public:
-  explicit CellVarProvider(std::shared_ptr<StateDescriptor> &sd) : state_(sd) {}
-  void AddPrivate(const std::string &package, const std::string &var,
+  explicit DenseFieldProvider(std::shared_ptr<StateDescriptor> &sd) : state_(sd) {}
+  void AddPrivate(const std::string &package, const std::string &label,
                   const Metadata &metadata) {
-    state_->AddField(package + "::" + var, metadata);
+    state_->AddFieldImpl(VarID(package + "::" + label), metadata);
   }
-  void AddProvides(const std::string &package, const std::string &var,
+  void AddProvides(const std::string & /*package*/, const std::string &label,
```

Comment:
Because `package` was unused and `label` is more descriptive than `var`, because we're not actually adding the `var` (shared pointer to `CellVariable`) but just the label.

Reply:
Fair enough.